tests/neg-custom-args/no-experimental/experimentalAnnotation.scala

```diff
@@ -0,0 +1,76 @@
+import scala.annotation.experimental
+
+@experimental // error
+val x = ()
+
+@experimental // error
+def f() = ()
+
+@experimental // error
+class A:
+  def f() = 1
+
+class B extends A: // error
+  override def f() = 2
+
+@experimental // error
+type X
+
+@experimental // error
+object X:
+  def fx() = 1 // error
+
+class C:
+  @experimental // error
+  def f() = 1
+
+class D extends C:
+  override def f() = 2
+
+trait A2:
+  @experimental // error
+  def f(): Int
+
+trait B2:
+  def f(): Int
+
+class C2 extends A2, B2:
+  def f(): Int = 1
+
+object Extractor1:
+  def unapply(s: Any): Option[A] = ??? // error
+
+object Extractor2:
+  @experimental // error
+  def unapply(s: Any): Option[Int] = ???
+
+def test(
+  p1: A, // error
+  p2: X, // error
+  p3: List[A], // error
+): Unit =
+  f() // error
+  x // error
+  new A // error
+  new B // error
+  X.fx() // error
+  import X.fx // error
+  fx() // error
+  val i1 = identity[X] // error // error
+  val i2 = identity[A] // error // error
+  val a: A = ??? // error
+  val b: B = ??? // error
+  val c: C = ???
+  val d: D = ???
+  val c2: C2 = ???
+  a.f() // error
+  b.f() // error
+  c.f() // error
+  d.f() // error
+  c2.f() // ok because B2.f is a stable API
+  ()
+
```

Comment:
We should check SAM-converted lambdas too:
```suggestion
@experimental
trait ExpSAM {
  def foo(x: Int): Int
}
def bar(f: ExpSam): Unit = {}
bar(x => x) // error: `x => x` has experimental type `ExpSam`
```